src/mainloop.rkt

```diff
@@ -141,39 +193,58 @@
 ; not all taylor transforms are valid in a given repr, return false on failure
 (define (taylor-expr expr repr var f finv)
   (define expr* (resugar-program expr repr #:full #f))
-  (with-handlers ([exn:fail? (const #f)]) 
+  (with-handlers ([exn:fail? (const #f)]) ; in case taylor fails
     (define genexpr (approximate expr* var #:transform (cons f finv)))
-    (λ () (desugar-program (genexpr) repr (*var-reprs*) #:full #f))))
+    (λ (x) (desugar-program (genexpr) repr (*var-reprs*) #:full #f))))
 
 (define (exact-min x y)
   (if (<= x y) x y))
 
 (define (much-< x y)
   (< x (/ y 2)))
 
+;; Taylor is problematic since it doesn't know what reprs are
+;; There are two types of errors that occur due to this inconsistency
+;;  - reduce:
+;;      the internal simplifier will try to desugar an subexpression
+;;      with an operator/precision mismatch
+;;  - external:
+;;      Taylor is successful in generating an expression but
+;;      external desugaring fails because of an unsupported/mismatched
+;;      operator
+(define (taylor-fail-desugaring expr)
+  (λ _
+    (debug #:from 'progress #:depth 5 "Series expansion (desugaring failure)")
+    (debug #:from 'progress #:depth 5 "Problematic expression: " expr)
```

Comment:
Have we grepped the logs for this? Ultimately we will want to fix things until this cannot occur and we're confident in that.

Reply:
Founds bugs in the other plugins! We need to fix softposit-herbie because it can't do taylor right now. This has more to do with the use of the `nonffi` field. I think this will be solved when we split the operator table. It's still possible that Taylor will fail if a representation doesn't support certain operators.